pytest_mypy_plugins/collect.py

```diff
@@ -86,26 +86,42 @@ def collect(self) -> Iterator["YamlTestItem"]:
 
             skip = self._eval_skip(str(raw_test.get("skip", "False")))
             if not skip:
-                yield YamlTestItem(
-                    name=test_name,
-                    collector=self,
-                    config=self.config,
-                    files=test_files,
-                    starting_lineno=starting_lineno,
-                    environment_variables=extra_environment_variables,
-                    disable_cache=disable_cache,
-                    expected_output_lines=output_from_comments + expected_output_lines,
-                    parsed_test_data=raw_test,
-                    mypy_config=additional_mypy_config,
-                )
+                try:
+                    yield YamlTestItem.from_parent(
+                        self,
+                        name=test_name,
+                        files=test_files,
+                        starting_lineno=starting_lineno,
+                        environment_variables=extra_environment_variables,
+                        disable_cache=disable_cache,
+                        expected_output_lines=output_from_comments + expected_output_lines,
+                        parsed_test_data=raw_test,
+                        mypy_config=additional_mypy_config,
+                    )
+                except AttributeError:
```

Comment:
I don't think that this legacy support is required. We can safely ignore it.
Because we can specify the minimal `pytest` version that already support this new way of doing things.